webpack/scenes/Settings/Tables/TableActions.js

```diff
@@ -0,0 +1,73 @@
+import { foremanApi, userId } from '../../../services/api';
+
+import {
+  TABLES_REQUEST,
+  TABLES_SUCCESS,
+  TABLES_FAILURE,
+  CREATE_TABLE,
+  CREATE_TABLE_SUCCESS,
+  CREATE_TABLE_FAILURE,
+  UPDATE_TABLE,
+  UPDATE_TABLE_SUCCESS,
+  UPDATE_TABLE_FAILURE,
+} from './TableConstants';
+
+const getResponseError = ({ data }) => data && (data.displayMessage || data.error);
+
+export const loadTables = () => (dispatch) => {
+  dispatch({ type: TABLES_REQUEST, params: {} });
+
+  return foremanApi
+    .get(`/users/${userId()}/table_preferences`, {})
+    .then(({ data }) => {
+      dispatch({
+        type: TABLES_SUCCESS,
+        payload: data,
+      });
+    })
+    .catch((result) => {
+      const { response } = result;
+      dispatch({
+        type: TABLES_FAILURE,
+        error: getResponseError(response),
```

Comment:
Could you use `payload` instead of `error`?
IMO we should only dispatch type and payload as a formal API for inner communication.